_overviews/scala3-macros/faq.md

```diff
@@ -23,3 +23,34 @@ x match
   case '{ $x: t } =>
     // `x: Expr[X & t]` where `t` is the precise type of `x`
 ```
+
+## How do I summon all types of a tuple type?
+If I have a type `(T1, T2, ...)` how do I generate the term for `(summon[T1], summon[T2], ...)` or get the individual expressions with the summoned values?
+
+Depending on you use case the way you will summon them will vary.
+The code you will need will depend on the kind of output you want (`Expr[Tuple]`, `List[Expr[Any]]`, or something else) and how you need errors to be reported.
+Here are two examples that should give you the basic skeleton for your variant of this code.
+
+```scala
+  def summonAllInList[T](using Type[T])(using Quotes): List[Expr[Any]] = {
+    Type.of[T] match
+      case '[ head *: tail ] =>
+        Expr.summon[head] match
+          case Some(headExpr) => headExpr :: Expr.summonAll[tail]
```

Comment:
is `Expr.summonAll` meant to be `summonAllInList`?

Reply:
My bad. It it supposed to be a recursive call to `summonAllInList`

Reply:
Fixed here and in the second example